### PR TITLE
[enhancement](memory) PODArray replaces MemPool in PredicateColumn

### PR DESCRIPTION
**Pull request by xinyiZzz**
# Proposed changes

Issue Number: close #xxx

## Problem summary

MemPool is about to be removed, replaced by Arena and PODArray.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by xinyiZzz** [comment]
run buildall

File: be/src/vec/columns/predicate_column.h [hunk 1]
```diff
@@ -547,7 +540,7 @@ class PredicateColumnType final : public COWHelper<IColumn, PredicateColumnType<
 private:
     Container data;
     // manages the memory for slice's data(For string type)
-    std::unique_ptr<MemPool> _pool;
+    PaddedPODArray<char> strings;
```
Comment (yiguolei) [inline be/src/vec/columns/predicate_column.h:543]: Chars

Comment (xinyiZzz) [reply be/src/vec/columns/predicate_column.h]: done

**Comment by xinyiZzz** [comment]
run buildall

File: be/src/vec/columns/predicate_column.h [hunk 2]
```diff
@@ -276,11 +275,10 @@ class PredicateColumnType final : public COWHelper<IColumn, PredicateColumnType<
             return;
         }
         if constexpr (std::is_same_v<T, StringRef>) {
-            if (_pool == nullptr) {
-                _pool.reset(new MemPool());
-            }
             const auto total_mem_size = offsets[num] - offsets[0];
-            char* destination = (char*)_pool->allocate(total_mem_size);
+            strings.resize(strings.size() + total_mem_size);
+            char* destination = reinterpret_cast<char*>(strings.end().base());
```
Comment (yiguolei) [inline be/src/vec/columns/predicate_column.h:280]: destination = strings.data() + strings.size(); will be more clear.

File: be/src/vec/columns/predicate_column.h [hunk 3]
```diff
@@ -300,16 +298,13 @@ class PredicateColumnType final : public COWHelper<IColumn, PredicateColumnType<
...
-            }
-
             size_t total_mem_size = 0;
             for (size_t i = 0; i < num; i++) {
                 total_mem_size += len_array[i];
             }
 
-            char* destination = (char*)_pool->allocate(total_mem_size);
+            strings.resize(strings.size() + total_mem_size);
+            char* destination = reinterpret_cast<char*>(strings.end().base());
```
Comment (yiguolei) [inline be/src/vec/columns/predicate_column.h:307]: the same here.

**Comment by xinyiZzz** [comment]
run buildall

**Comment by xinyiZzz** [comment]
run buildall

**Review by yiguolei — DISMISSED** [review]
LGTM

**Comment by xinyiZzz** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**